### DISCUSSION TITLE
Is there any example that use zustand without React with middleware persist and immer?

### DISCUSSION THREAD

**Discussion by cuongdo926** [pmndrs/zustand#1545] [General] [answered]
Zustand works very well using hooks. Currently, I create a store that uses middlewares persist and immer. 
But I have a problem with react-router v6. I want to use a store in the loader that does not allow a hook there.  
How can I create a store that reuses my code but still work outside react (in the router loader)?

**Comment by dai-shi** [comment, marked as answer]
If you create a hook `useFooStore`, you can read the state outside react with`useFooStore.getState()`.
